src/NumericInput.js

```diff
@@ -26,18 +28,18 @@ export default class NumericInput extends Component {
     }
 
     onStepUp = () => {
-        const { step, onChange } = this.props;
+        const { step, decimal, onChange } = this.props;
         const { value } = this.state;
-        const newValue = +value + step;
-        this.setState({ value: newValue});
+        const newValue = +(+value + step).toFixed(decimal);
+        this.setState({ value: newValue });
         onChange({ target: { value: newValue } });
     }
 
     onStepDown = () => {
-        const { step, onChange } = this.props;
+        const { step, onChange, decimal } = this.props;
         const { value } = this.state;
-        const newValue = +value - step;
-        this.setState({ value: newValue});
+        const newValue = +(+value - step).toFixed(decimal);
```

Comment:
The + sign outside does not seem needed?
The +value - step is sure to be number if step is number, and if itsn't there is a bigger problem :)